haproxy-ingress/templates/controller-servicemonitor.yaml

```diff
@@ -0,0 +1,30 @@
+{{- if and .Values.controller.serviceMonitor.enabled .Values.controller.metrics.enabled (or .Values.controller.stats.enabled .Values.controller.metrics.embedded) -}}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+{{- if .Values.controller.serviceMonitor.annotations }}
+  annotations:
+    {{- toYaml .Values.controller.serviceMonitor.annotations | nindent 4 }}
+{{- end }}
+  labels:
+    {{- include "haproxy-ingress.labels" . | nindent 4 }}
+{{- if .Values.controller.serviceMonitor.labels }}
+    {{ toYaml .Values.controller.serviceMonitor.labels | nindent 4 }}
```

Comment:
add a trailing `{{-` here to avoid a blank line in the final yaml

Reply:
Fixed blank line